### PR TITLE
Fix compatibility issue with SponsorBlock

### LINKED ISSUE #3377: YouTube: SponsorBlock popups are detected as artist/track names
#### Describe the bug

When a [SponsorBlock](https://sponsor.ajay.app) prompt appears in the YouTube player (asking to skip a section of a type that's set to manually skip), the text in the popup is parsed as an artist and title, and the currently-playing song is replaced by that text.

This also breaks data corrections -- having replaced the captured text the first time with the song playing at the time, other songs which trigger the same skips end up being reported as the original song.

#### How to reproduce

1. Install SponsorBlock and set the skips to manual:
  Non-Music Section set to Manual Skip
2. Go to a music video with a segment at the start or end that SponsorBlock recognises (e.g. [this one](https://youtu.be/EVBsypHzF3U)).

* When a "Non-Music" popup appears, the detected song is "Music" by "Non".
* When a "Endcards/Credits" popup appears, the detected song changes to "Credits" by "Endcards".

#### Expected behaviour

SponsorBlock popups shouldn't be parsed when looking for the song title and artist name.

#### Screenshots

These are the in-player popups:
Non-Music popup Credits popup

And the resulting song detection:
Non by Music Credits by Endcards

#### Debug logs

<details>
<summary>Logs</summary>

```
Tab 192: Song unprocessed: {
  "parsed": {
    "track": "Music",
    "artist": "Non",
    "album": null,
    "albumArtist": null,
    "uniqueID": null,
    "duration": null,
    "currentTime": null,
    "isPlaying": false,
    "trackArt": null,
    "isPodcast": false,
    "originUrl": "https://youtu.be/EVBsypHzF3U"
  },
  "processed": {
    "track": null,
    "album": null,
    "artist": null,
    "albumArtist": null,
    "duration": null
  },
  "flags": {
    "isScrobbled": false,
    "isCorrectedByUser": true,
    "isValid": true,
    "isMarkedAsPlaying": true,
    "isSkipped": false,
    "isReplaying": false
  },
  "metadata": {
    "startTimestamp": 1663977452,
    "label": "YouTube",
    "trackArtUrl": "http://coverartarchive.org/release/71babf1e-82cc-4c40-be2e-df69c38f2784/front-500",
    "notificationId": "6"
  },
  "connectorLabel": "YouTube"
}

Tab 192: Clearing playback timer destination time

Send "get info" request: 1

Tab 192: Song finished processing: {
  "parsed": {
    "track": "Music",
    "artist": "Non",
    "album": null,
    "albumArtist": null,
    "uniqueID": null,
    "duration": null,
    "currentTime": null,
    "isPlaying": false,
    "trackArt": null,
    "isPodcast": false,
    "originUrl": "https://youtu.be/EVBsypHzF3U"
  },
  "processed": {
    "track": null,
    "album": null,
    "artist": null,
    "albumArtist": null,
    "duration": null
  },
  "flags": {
    "isScrobbled": false,
    "isCorrectedByUser": false,
    "isValid": true,
    "isMarkedAsPlaying": false,
    "isSkipped": false,
    "isReplaying": false
  },
  "metadata": {
    "startTimestamp": 1663977496,
    "label": "YouTube"
  },
  "connectorLabel": "YouTube"
}
```
</details>

#### Environment

- OS: Windows 10
- Browser: Firefox 106 (developer edition)
- Extension version: v2.70.1

#### Additional context

SponsorBlock version: 5.0.4

### PR DESCRIPTION
**Pull request by yayuyokitano**
SponsorBlock seems to have recently made a change that hijacks the youtube chapter title element.
This causes web scrobbler to catastrophically fail to recognize video titles.
This PR attempts to reduce the issue as smoothly as possible by detecting the presence of a SponsorBlock-added element, and if so, ignoring certain pieces of text in the chapter title element.
Fixes #3377

### CODE REVIEW

File: src/connectors/youtube.js [hunk 1]
```diff
@@ -140,7 +140,16 @@ function setupEventListener() {
 }
 
 function areChaptersAvailable() {
-	return Util.getTextFromSelectors(chapterNameSelector);
+	const text = Util.getTextFromSelectors(chapterNameSelector);
+
+	// SponsorBlock extension hijacks chapter element. Ignore it.
+	const sponsorBlockButton = document.getElementById('startSegmentButton');
+	if (sponsorBlockButton && (text === 'Non-Music' || text === 'Sponsor' || text === 'Credits/Endcards' || text === 'Intermission')) {
```
Comment (inverse) [inline src/connectors/youtube.js:147]:
```suggestion
	if (sponsorBlockButton && (text === 'Non-Music' || text === 'Sponsor' || text === 'Endcards/Credits' || text === 'Intermission')) {
```

Should it be this way round?


See: https://github.com/ajayyy/SponsorBlock/blob/1a0dd19cf0feff639e2e7943736409278c5fbee6/public/_locales/en/messages.json#L671

**Review by inverse — APPROVED** [review]
Looks good :+1: 

Wonder if we need to extend it in the future to support non-en locales.

See: https://github.com/ajayyy/SponsorBlock/tree/1a0dd19cf0feff639e2e7943736409278c5fbee6/public/_locales

File: src/connectors/youtube.js (hunk 1 above)
Comment (yayuyokitano) [reply src/connectors/youtube.js]: Yep, my bad

**Comment by yayuyokitano** [comment]
> Looks good 👍
> 
> Wonder if we need to extend it in the future to support non-en locales.
> 
> See: https://github.com/ajayyy/SponsorBlock/tree/1a0dd19cf0feff639e2e7943736409278c5fbee6/public/_locales

Ughh yeah we probably should eventually

**Comment by ajayyy** [comment]
I added a class when there it is displaying non-chapter text

Check for `sponsorBlock-segment-title` on `.ytp-chapter-title-content` element

https://github.com/ajayyy/SponsorBlock/commit/4a532e751cc324227cdb6865a2247bc5556f0238

**Comment by yayuyokitano** [comment]
> I added a class when it is displaying non-chapter text
> 
> Check for `sponsorBlock-segment-title` on `.ytp-chapter-title-content` element
> 
> [ajayyy/SponsorBlock@4a532e7](https://github.com/ajayyy/SponsorBlock/commit/4a532e751cc324227cdb6865a2247bc5556f0238)

Cool, thank you! I'll have a look into it tmr

**Comment by yayuyokitano** [comment]
@ajayyy opened #3422 using the class you provided. Please do let us know when it is released to production, as I won't be checking constantly

**Pull request merged**